From a949f1de2d7201de603e2f4fc66ed75636e39d80 Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Sat, 23 Mar 2024 16:34:50 -0300
Subject: [PATCH 1/2] build: Appease MSYS2 UCRT64 GCC 13

Undefining this macro makes GCC in standards C++ mode very unhappy:

In file included from D:/msys64/ucrt64/include/c++/13.2.0/bits/requires_hosted.h:31,
                 from D:/msys64/ucrt64/include/c++/13.2.0/string:38,
                 from ..\subprojects\webrtc-audio-processing\webrtc/rtc_base/system/file_wrapper.h:17,
                 from ../subprojects/webrtc-audio-processing/webrtc/rtc_base/system/file_wrapper.cc:11:
D:/msys64/ucrt64/include/c++/13.2.0/x86_64-w64-mingw32/bits/c++config.h:666:2: warning: #warning "__STRICT_ANSI__ seems to have been undefined; this is not supported" [-Wcpp]
  666 | #warning "__STRICT_ANSI__ seems to have been undefined; this is not supported"
      |  ^~~~~~~

See: https://github.com/fmtlib/fmt/issues/2059#issue-761209068

See: #32
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 3843e92..de07ded 100644
--- a/meson.build
+++ b/meson.build
@@ -92,7 +92,7 @@ elif host_system == 'linux'
   os_deps += [dependency('threads')]
   have_posix = true
 elif host_system == 'windows'
-  platform_cflags += ['-DWEBRTC_WIN', '-D_WIN32', '-U__STRICT_ANSI__']
+  platform_cflags += ['-DWEBRTC_WIN', '-D_WIN32']
   # this one is for MinGW to get format specifiers from inttypes.h in C++
   platform_cflags += ['-D__STDC_FORMAT_MACROS=1']
   # Avoid min/max from windows.h which breaks std::min/max
-- 
2.44.0.windows.1

